### PR TITLE
merging back to feature branch

### PR DESCRIPTION
**Pull request by devights**

### CODE REVIEW

File: myuw/dao/myuw_notice.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        if not(notice.is_alumni or notice.is_applicant or notice.is_grad or
+                notice.is_grad_c2 or notice.is_pce or notice.is_student or
+                notice.is_undergrad or notice.is_undergrad_c2 or
+                notice.is_fyp or notice.is_past_student or
+                notice.is_clinician or notice.is_employee or
+                notice.is_faculty or notice.is_instructor or
+                notice.is_past_employee or notice.is_retiree or
+                notice.is_staff_employee or notice.is_stud_employee):
+            user_notices.append(notice)
+        if notice.is_alumni and affiliations["alumni"]:
```
Comment (ezturner) [inline myuw/dao/myuw_notice.py]: Could you do something here with a dynamic lookup, like:

for key in affiliations:
&nbsp;&nbsp;&nbsp;&nbsp;if getattr(notice, "is_" + key):
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;user_notices.append(notice)

File: myuw/dao/myuw_notice.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    for notice in campus_notices:
+        # If affiliation not set show to all affiliations
+        if not(notice.is_alumni or notice.is_applicant or notice.is_grad or
+                notice.is_grad_c2 or notice.is_pce or notice.is_student or
+                notice.is_undergrad or notice.is_undergrad_c2 or
+                notice.is_fyp or notice.is_past_student or
+                notice.is_clinician or notice.is_employee or
+                notice.is_faculty or notice.is_instructor or
+                notice.is_past_employee or notice.is_retiree or
+                notice.is_staff_employee or notice.is_stud_employee):
```
Comment (ezturner) [inline myuw/dao/myuw_notice.py]: Could you refactor this into a model method that scans all children and checks if any of the children/variables prefixed with is_* is set to True?

File: myuw/dao/myuw_notice.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        if notice.is_undergrad and affiliations["undergrad"]:
+            user_notices.append(notice)
+            pass
+        if notice.is_undergrad_c2 and affiliations["undergrad_c2"]:
+            user_notices.append(notice)
+            pass
+        if notice.is_fyp and affiliations["fyp"]:
+            user_notices.append(notice)
+            pass
+        if notice.is_past_student and affiliations["past_stud"]:
```
Comment (ezturner) [inline myuw/dao/myuw_notice.py]: This'd have to be renamed to is_past_stud (here and in the metohd) to match the above suggestion

File: myuw/dao/myuw_notice.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            user_notices.append(notice)
+        if notice.is_alumni and affiliations["alumni"]:
+            user_notices.append(notice)
+            pass
+        if notice.is_applicant and affiliations["applicant"]:
+            user_notices.append(notice)
+            pass
+        if notice.is_grad and affiliations["grad"]:
+            user_notices.append(notice)
+            pass
```
Comment (ezturner) [inline myuw/dao/myuw_notice.py]: Could you do something here with a dynamic lookup, like:

for key in affiliations:
&nbsp;&nbsp;&nbsp;&nbsp;if getattr(notice, "is_" + key):
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;user_notices.append(notice)

File: myuw/dao/myuw_notice.py (hunk 2 above)
Comment (devights) [reply myuw/dao/myuw_notice.py]: Not without refactoring the campus attrs so I can ignore them when scanning the is_ affil attrs.

Comment (ezturner) [reply myuw/dao/myuw_notice.py]: Ah, that's fair.

**Pull request merged**